### PR TITLE
authorized users: optionally allow edit/delete of findings

### PR DESCRIPTION
**Pull request by valentijnscholten**
There are plans for a (design) discussion about RBAC, but as we have seen that is not an easy / small task. The biggest hurdle currently when rolling out DD inside an organisation is that teams (`authorized_users`) cannot edit their own findings.

This PR implements a quick win for this. I tried to use some of the parts of #2414, so we don't have to throw everything away later. Turns out Django is not really good with object level permissions, unless you create an entry in a permission table for every user or group that is allowed to access a finding. Which is not scaleable. This PR:

- adds a decoratoed `@user_is_authorized` to decorate functions to check for the correct permission.
- adds two `settings.dist.py` entries to control if authorized users are allowed more than just viewing findings:
```
AUTHORIZED_USERS_ALLOW_CHANGE = env('DD_AUTHORIZED_USERS_ALLOW_CHANGE')
AUTHORIZED_USERS_ALLOW_DELETE = env('DD_AUTHORIZED_USERS_ALLOW_DELETE')
```
- both default to False
- updates dropdown menus when viewing a finding, a test or a list of findings
- makes the above also work with bulk edit / bulk delete
- fixes some bugs around datatables for users that are not staff users

### CODE REVIEW

File: dojo/templates/dojo/view_test.html [hunk 1]
```diff
@@ -321,98 +327,112 @@ <h4 class="has-filters"> Findings ({{findings_count}}) <small>{{ test.id|get_sev
...
+                                           <a href="{% url 'action_history' finding|content_type finding.id %}">
+                                               <i class="fa fa-history"></i> View History
+                                           </a>
+                                       </li>
+                                       {% if user|is_authorized_for_delete:finding %}
+                                       <li>
+                                         <form method="post" action="{% url 'delete_finding' finding.id %}"
+                                              style="display: inline" class="form-inline form" id="delete-finding-menu-{{ finding.id }}-form">
+                                            {% csrf_token %}
+                                            <input type="hidden" name="return_url" value="{{ request.get_full_path }}" />
```
[the remark this answers is not in the document]
Comment (valentijnscholten) [reply dojo/templates/dojo/view_test.html:421]: hidden fields....

File: dojo/templates/dojo/view_test.html [hunk 2]
```diff
@@ -321,98 +327,112 @@ <h4 class="has-filters"> Findings ({{findings_count}}) <small>{{ test.id|get_sev
...
+                                            <input type="hidden" name="id" value="{{ finding.id }}"/>
```
[the remark this answers is not in the document]
Comment (valentijnscholten) [reply dojo/templates/dojo/view_test.html:422]: hidden hidden hidden

**Comment by madchap** [comment]
I am so glad to see this!

I was wondering while giving it a whirl whether if we should extend the product's menu to have 

image

and

image

if users are allowed to `CHANGE`?

**Comment by valentijnscholten** [comment]
Could be done, but do you trust your users to understand the (dedupe) implications of uploading something? Or do a re-upload which may close findings that are not in the uploaded report?
We could do another flag, but then we're basically implementing RBAC, but in a poor way :-)

**Comment by madchap** [comment]
> Could be done, but do you trust your users to understand the (dedupe) implications of uploading something? Or do a re-upload which may close findings that are not in the uploaded report?
> We could do another flag, but then we're basically implementing RBAC, but in a poor way :-)

Right. Let's keep it "sweet and simple" so to speak ;-)

**Pull request merged**